### PR TITLE
bump sphinx

### LINKED ISSUE #364: Install instructions out of date
Following the directions in the readme produces the following error. 

`pip3 uninstall sphinx==3.5.0 breathe==4.28.0 sphinx_rtd_theme sphinxcontrib-plantuml jinja2==3.0.3
make html`

`Traceback (most recent call last):
  File "/home/josh/.local/bin/sphinx-build", line 5, in <module>
    from sphinx.cmd.build import main
  File "/home/josh/.local/lib/python3.10/site-packages/sphinx/cmd/build.py", line 25, in <module>
    from sphinx.application import Sphinx
  File "/home/josh/.local/lib/python3.10/site-packages/sphinx/application.py", line 32, in <module>
    from sphinx.config import Config
  File "/home/josh/.local/lib/python3.10/site-packages/sphinx/config.py", line 23, in <module>
    from sphinx.util import logging
  File "/home/josh/.local/lib/python3.10/site-packages/sphinx/util/__init__.py", line 35, in <module>
    from sphinx.util import smartypants  # noqa
  File "/home/josh/.local/lib/python3.10/site-packages/sphinx/util/smartypants.py", line 33, in <module>
    from sphinx.util.docutils import __version_info__ as docutils_version
  File "/home/josh/.local/lib/python3.10/site-packages/sphinx/util/docutils.py", line 31, in <module>
    from sphinx.util.typing import RoleFunction
  File "/home/josh/.local/lib/python3.10/site-packages/sphinx/util/typing.py", line 34, in <module>
    from types import Union as types_Union
ImportError: cannot import name 'Union' from 'types' (/usr/lib/python3.10/types.py)
`

The issue is discussed [here](https://github.com/sphinx-doc/sphinx/issues/9562). 

In order to install correctly, with python 3.10 I did the following: 

``pip3 uninstall sphinx==4.2.0 breathe sphinx_rtd_theme sphinxcontrib-plantuml jinja2==3.0.3
make html``

### PR DESCRIPTION
**Pull request by jwallace42**
Pr to resolve #364.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -17,7 +17,7 @@ references:
         apt install openjdk-8-jre -y
         apt install bash git openssh-server -y
         apt install python3-pip python3 -y
-        pip3 install sphinx==3.5.0 docutils==0.14 sphinx_rtd_theme breathe==4.28.0 sphinxcontrib-plantuml jinja2==3.0.3
+        pip3 install sphinx==4.2.0 breathe sphinx_rtd_theme sphinxcontrib-plantuml jinja2==3.0.3
```
Comment (SteveMacenski) [inline .circleci/config.yml:20]: Version control breathe

File: .circleci/config.yml [hunk 2]
```diff
@@ -17,7 +17,7 @@ references:
         apt install openjdk-8-jre -y
         apt install bash git openssh-server -y
         apt install python3-pip python3 -y
-        pip3 install sphinx==3.5.0 docutils==0.14 sphinx_rtd_theme breathe==4.28.0 sphinxcontrib-plantuml jinja2==3.0.3
```
Comment (SteveMacenski) [inline .circleci/config.yml:20]: Do we not need docutils?

**Review by SteveMacenski — CHANGES_REQUESTED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (SteveMacenski) [reply .circleci/config.yml:20]: I don't want updates to break CI

File: .circleci/config.yml (hunk 2 above)
Comment (jwallace42) [reply .circleci/config.yml:20]: It is a dependency of breathe. I added it back in to be explicit.

**Comment by SteveMacenski** [comment]
https://output.circle-artifacts.com/output/job/7793d1a2-8e0a-423d-835c-63cb3e869c5f/artifacts/0/html/build_instructions/index.html

It looks like this broke the collapsing sections in build/install (ref: https://navigation.ros.org/build_instructions/index.html)

**Pull request closed without merging**